src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
+        auto result = ripplePathFind(
+            std::make_shared<RippleLineCache>(
+                ledger), src.pk(), dest.pk(), saDstAmount,
+            ledger, jvSrcCurrencies, contextPaths, level);
+        if(!result.first)
+            throw std::runtime_error(
+            "Path_test::findPath: ripplePathFind find failed");
+
+        return result.second;
+    }
+
     void
     test_no_direct_path_no_intermediary_no_alternatives()
     {
         testcase("no direct path no intermediary no alternatives");
-        std::uint64_t const xrp = std::mega::num;
-
-        auto master = createAccount("masterpassphrase", KeyType::ed25519);
-
-        std::shared_ptr<Ledger const> LCL;
-        Ledger::pointer ledger;
-        std::tie(LCL, ledger) = createGenesisLedger(100000 * xrp, master);
-
-        auto accounts = createAndFundAccounts(master, { "alice", "bob" }, 
-            KeyType::ed25519, 10000 * xrp, ledger);
-        auto& alice = accounts["alice"];
-        auto& bob = accounts["bob"];
-        expectNotEquals(alice.pk.humanAccountID(), master.pk.humanAccountID());
-        
-        auto alternatives = findPath(ledger, alice, bob, { Currency("USD") },
-            Amount(5, "USD", alice), log.stream());
-        log << "ripplePathFind alternatives: " << alternatives;
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob");
 
-        expectEquals(alternatives.size(), 0);
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"](0)}, Account("bob")["USD"](5));
+        expect(alternatives.size() == 0);
     }
 
     void
     test_direct_path_no_intermediary()
     {
         testcase("direct path no intermediary");
-        std::uint64_t const xrp = std::mega::num;
-
-        auto master = createAccount("masterpassphrase", KeyType::ed25519);
-
-        std::shared_ptr<Ledger const> LCL;
-        Ledger::pointer ledger;
-        std::tie(LCL, ledger) = createGenesisLedger(100000 * xrp, master);
-
-        // Create accounts
-        auto accounts = createAndFundAccounts(master, { "alice", "bob" },
-            KeyType::ed25519, 10000 * xrp, ledger);
-        auto& alice = accounts["alice"];
-        auto& bob = accounts["bob"];
-        expectNotEquals(alice.pk.humanAccountID(), master.pk.humanAccountID());
-
-        // Set credit limit
-        trust(bob, alice, "USD", 700, ledger);
-
-        // Find path from alice to bob
-        auto alternatives = findPath(ledger, alice, bob, { Currency("USD") },
-            Amount(5, "USD", bob), log.stream());
-        log << "ripplePathFind alternatives: " << alternatives;
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob");
+        env.trust(Account("alice")["USD"](700), "bob");
+
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"](0)}, Account("bob")["USD"](5));
+        Json::Value jv;
+        Json::Reader().parse(R"([{
```

Comment:
Maybe create this directly instead of through a string? I haven't used our `Json::Value` that much, so I don't know if that would work out better or not, but consider it.